src/microbe_stage/editor/upgrades/FlagellumUpgradeGUI.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Godot;
+
+/// <summary>
+///   Upgrade GUI for the flagellum that allows changing its length
+/// </summary>
+public partial class FlagellumUpgradeGUI : VBoxContainer, IOrganelleUpgrader
+{
+#pragma warning disable CA2213
+    [Export]
+    private Slider lengthSlider = null!;
+#pragma warning restore CA2213
+
+    public override void _Ready()
+    {
+    }
+
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame, float costMultiplier)
+    {
+        if (organelle.Upgrades?.CustomUpgradeData is FlagellumUpgrades flagellumUpgrades)
```

Comment:
Upgrade GUIs aren't reused (right now) but you should still put in the logic that the slider is reset to the default value if there are no flagellum upgrades specified.

Reply:
And this still looks like it is missing the `else` branch.